src/tests/ftest/util/dmg_utils.py

```diff
@@ -102,20 +102,109 @@ def network_scan(self, provider=None, all_devs=False):
+        0000:5e:00.0 INTEL SSDPE2KE016T8  VDV10170    0         1.6 TB
+        0000:5f:00.0 INTEL SSDPE2KE016T8  VDV10170    0         1.6 TB
+        0000:81:00.0 INTEL SSDPED1K750GA  E2010475    1         750 GB
+        0000:da:00.0 INTEL SSDPED1K750GA  E2010475    1         750 GB
+        """
+
+        # Sample dmg storage scan output. Don't delete this sample because it
+        # helps to develop and debug the regex.
+        """
+         Hosts    SCM Total             NVMe Total
+        -----    ---------             ----------
+        wolf-130 6.4 TB (2 namespaces) 4.7 TB (4 controllers)
+        """
+
+        data = {}
+
+        if verbose:
+            vals = re.findall(
+                r"--------\n([a-z0-9-]+)\n--------|"
+                r"\n([a-z0-9_]+)[ ]+([\d]+)[ ]+([\d.]+) ([A-Z]+)|"
+                r"([a-f0-9]+:[a-f0-9]+:[a-f0-9]+.[a-f0-9]+)[ ]+"
+                r"(\S+)[ ]+(\S+)[ ]+(\S+)[ ]+(\d+)[ ]+([\d.]+)"
+                r"[ ]+([A-Z]+)[ ]*\n", self.result.stdout)
+
+            data = {}
+            host = vals[0][0]
+            data[host] = {}
+            data[host]["scm"] = {}
+            i = 1
+            while i < len(vals):
+                if vals[i][1] == "":
+                    break
+                pmem_name = vals[i][1]
+                socket_id = vals[i][2]
+                capacity = "{} {}".format(vals[i][3], vals[i][4])
+                data[host]["scm"][pmem_name] = {}
+                data[host]["scm"][pmem_name]["socket"] = socket_id
+                data[host]["scm"][pmem_name]["capacity"] = capacity
+                i += 1
+
+            data[host]["nvme"] = {}
+            while i < len(vals):
+                pci_addr = vals[i][5]
+                model = "{} {}".format(vals[i][6], vals[i][7])
+                fw_revision = vals[i][8]
+                socket_id = vals[i][9]
+                capacity = "{} {}".format(vals[i][10], vals[i][11])
+                data[host]["nvme"][pci_addr] = {}
+                data[host]["nvme"][pci_addr]["model"] = model
+                data[host]["nvme"][pci_addr]["fw_revision"] = fw_revision
+                data[host]["nvme"][pci_addr]["socket"] = socket_id
+                data[host]["nvme"][pci_addr]["capacity"] = capacity
+                i += 1
+
+        else:
+            vals = re.findall(
+                r"([a-z0-9-\[\]]+)\s+([\d.]+)\s+([A-Z]+)\s+"
+                r"\(([\w\s]+)\)\s+([\d.]+)\s+([A-Z]+)\s+\(([\w\s]+)",
+                self.result.stdout)
+            self.log.info("--- Non-verbose output parse result ---")
+            self.log.info(vals)
+
+            data = {}
+            for row in vals:
+                host = row[0]
+                data[host] = {
+                    "scm": {"capacity": None, "details": None},
+                    "nvme": {"capacity": None, "details": None}}
+                data[host]["scm"]["capacity"] = "{} {}".format(row[1], row[2])
```

Comment:
Optional, this could be simplified (also aligns with the first bullet in https://www.python.org/dev/peps/pep-0008/#programming-recommendations):
```suggestion
                data[host]["scm"]["capacity"] = " ".join(row[1:3])
```

Reply:
Done. Looks like this page has many useful resources.